dd-java-agent/agent-logging/src/main/java/datadog/trace/logging/LogReporter.java

```diff
@@ -0,0 +1,122 @@
+package datadog.trace.logging;
+
+import static java.nio.file.Files.readAllBytes;
+
+import datadog.trace.api.Config;
+import datadog.trace.api.flare.TracerFlare;
+import java.io.File;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.zip.ZipOutputStream;
+
+public class LogReporter implements TracerFlare.Reporter {
+  private static final LogReporter INSTANCE = new LogReporter();
+  private static final int MAX_LOGFILE_SIZE_MB = 15;
+  public static final int MAX_LOGFILE_SIZE_BYTES = MAX_LOGFILE_SIZE_MB << 20;
+  private Path capturedLogPath;
+  private boolean isFlarePrepared;
+  private static File outputConfiguredFile;
+  private static PrintStreamWrapper outputPrintStreamWrapper;
+
+  public LogReporter() {}
```

Comment:
Make this `private` to stop people from accidentally creating other instances.
```suggestion
  private LogReporter() {}
```